generator/osm2type.cpp

```diff
@@ -343,12 +367,30 @@ void MatchTypes(OsmElement * p, FeatureBuilderParams & params, function<bool(uin
 
   LeaveLongestTypes(matchedTypes);
 
-  for (auto const & path : matchedTypes)
+  auto const & cl = classif();
+
+  bool highwayAdded = false;
+  for (size_t i = 0; i < matchedTypes.size(); ++i)
   {
-    uint32_t const t = classif().GetTypeByPath(path);
+    auto const & path = matchedTypes[i];
+    uint32_t const type = cl.GetTypeByPath(path);
+    if (!filterType(type))
+      continue;
+
+    // Additional highway strict check.
+    // "busway" and "service" can be matched together, keep busway only.
+    if (path[0] == "highway")
+    {
+      if (highwayAdded)
```

Comment:
there are many cases ATM when we have multiple highway-* types per feature,
e.g. there could be `highway-path-bridge highway-path-bicycle highway-path-horse`

Reply:
Ops, I did it before reverting uniqueness, thanks